### PR TITLE
feat(card): Add component tokens

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #7180 

## Summary

Adds the following public component css properties:
`--calcite-card-background-color`: Specifies the background color of the component.
`--calcite-card-border-color`: Specifies the border color of the component.
`--calcite-card-box-shadow`: Specifies the box shadow of the component.
`--calcite-card-corner-radius`: Specifies the corner radius of the component.
`--calcite-card-subtitle-color`: Specifies the color of the component's `"subtitle"` slot.
`--calcite-card-title-color`: Specifies the color of the component's `"title"` slot.

Adds a Chromatic test and a demo page example. Also replaces broken image links in local demo.

### CODE REVIEW

File: packages/calcite-components/src/demos/card.html [hunk 1]
```diff
@@ -28,8 +28,14 @@
         padding: 0 18px;
       }
 
-      .rounded {
-        --calcite-border-radius-base: 5px;
+      .themed {
```
Comment (alisonailea) [inline packages/calcite-components/src/demos/card.html:31]: Thanks for adding demos!! 🎉

File: packages/calcite-components/src/components/card/card.scss [hunk 2]
```diff
@@ -1,87 +1,114 @@
+@import "~@esri/calcite-design-tokens/dist/scss/core";
+@import "~@esri/calcite-design-tokens/dist/scss/global";
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:2]: you shouldn't have to import this

File: packages/calcite-components/src/components/card/card.scss [hunk 3]
```diff
@@ -1,87 +1,114 @@
...
+*/
+
 :host {
   @apply block max-w-full;
-  & .calcite-card-container {
-    @apply bg-foreground-1
-    border-color-2
-    text-color-3
-    relative
+  --calcite-internal-card-background-color: var(--calcite-card-background-color, var(--calcite-color-foreground-1));
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:32]: I don't think we need to duplicate CSS vars for each token just to have an "-internal-". Just apply the component token directly.

File: packages/calcite-components/src/components/card/card.scss [hunk 4]
```diff
@@ -1,87 +1,114 @@
...
-.header,
-.footer {
-  @apply flex px-3 pt-3 pb-1;
+  @apply absolute inset-0 flex items-center;
 }
 
 .header {
-  @apply flex-col;
+  @apply flex flex-col;
+  padding-inline: $calcite-spacing-md;
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:70]: apply semantic tokens directly here 
`padding-inline: var(--calcite-spacing-md)`

File: packages/calcite-components/src/components/card/card.scss [hunk 5]
```diff
@@ -1,87 +1,114 @@
...
-    text-n2-wrap
-    p-3;
+  @apply text-n2-wrap;
+  color: var(--calcite-color-text-3);
+  padding: $calcite-spacing-md;
 }
 
 :host([selected]) .calcite-card-container {
-  @apply border-color-brand;
+  border-color: var(--calcite-color-brand);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:89]: replace this with `--calcite-card-container-border-color: var(--calcite-color-brand);`

File: packages/calcite-components/src/components/card/card.scss [hunk 6]
```diff
@@ -1,87 +1,114 @@
...
+  margin: 0;
+  font-weight: $calcite-font-weight-medium;
+  color: var(--calcite-internal-card-title-color);
 }
 
 :host([selectable]) {
   .header {
     // prevents overlap with checkbox (default header padding + no-overlap spacing)
-    padding-inline-end: theme("spacing.8");
+    padding-inline-end: $calcite-spacing-xxxl;
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:100]: replace with semantic CSS token

**Review by alisonailea — CHANGES_REQUESTED** [review]
Looks pretty good!
I think we can reduce overhead by reducing internal variables and using Semantic CSS tokens instead of SCSS

File: packages/calcite-components/src/components/card/card.scss [hunk 7]
```diff
@@ -1,87 +1,111 @@
...
     justify-between
-    border
-    border-solid
-    shadow-none
     duration-150
     ease-in-out
     overflow-hidden;
-    border-radius: var(--calcite-border-radius-base);
-  }
+  border: 1px solid var(--calcite-internal-card-border-color);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/card/card.scss:46]: `border:  var(--calcite-border-width-sm) solid var(--calcite-card-border-color);`

File: packages/calcite-components/src/components/card/card.scss (hunk 2 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss]: Removed.

File: packages/calcite-components/src/components/card/card.scss (hunk 3 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss]: Removed.

File: packages/calcite-components/src/components/card/card.scss (hunk 5 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss]: Updated 👍

File: packages/calcite-components/src/components/card/card.scss [hunk 8]
```diff
@@ -1,87 +1,92 @@
...
 }
 
 .container {
   @apply relative flex flex-auto flex-col;
 }
 
 :host([loading]) .calcite-card-container *:not(calcite-loader):not(.calcite-card-loader-container) {
-  @apply pointer-events-none opacity-0;
+  @apply pointer-events-none;
+  opacity: $calcite-opacity-0;
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:39]: @alisonailea leaving this for now, need to add a token here.

**Pull request closed without merging**